driver-core/src/main/java/com/datastax/driver/core/Connection.java

```diff
@@ -505,10 +505,8 @@ void setKeyspace(String keyspace) throws ConnectionException {
                 return existingAttempt.future;
             } else if (!existingAttempt.future.isDone()) {
                 // If the existing attempt is still in flight, fail this attempt immediately.
-                logger.warn("{} There is already an in progress attempt to set keyspace on this connection for keyspace '{}', aborting set keyspace for '{}'.  "
-                                + "This can happen if you constantly change the keyspace at runtime.",
-                        this, existingAttempt.keyspace, keyspace);
-                ksFuture.setException(new DriverException("Aborting attempt to set keyspace to '" + keyspace + "' since there is already an in flight attempt to set keyspace to '" + existingAttempt.keyspace + "'."));
+                ksFuture.setException(new DriverException("Aborting attempt to set keyspace to '" + keyspace + "' since there is already an in flight attempt to set keyspace to '" + existingAttempt.keyspace + "'.  "
+                        + "This can happen if you constantly change the keyspace at runtime."));
```

Comment:
Suggestion: "This can happen if you try to USE different keyspaces from the same session simultaneously."